control.go

```diff
@@ -166,13 +166,11 @@ func (c *controlConn) shuffleDial(endpoints []*HostInfo) (*Conn, error) {
 	// node.
 	shuffled := shuffleHosts(endpoints)
 
-	cfg := *c.session.connCfg
-	cfg.disableCoalesce = true
-
-	var err error
+	var (
+		err  error
+		conn *Conn
+	)
 	for _, host := range shuffled {
-		var conn *Conn
-		c.session.dial(host, &cfg, c)
```

Comment:
yeah this looks very wrong, I think it should just be

```
conn, err = c.session.dial(host, &cfg, c)
```

Reply:
@Zariel you mean keep the `dial` call but remove the `connect` one?